LibCarla/source/carla/streaming/detail/tcp/ServerSession.cpp

```diff
@@ -126,7 +138,6 @@ namespace tcp {
   }
 
   void ServerSession::CloseNow() {
-    DEBUG_ASSERT(_strand.running_in_this_thread());
     _deadline.cancel();
     if (_socket.is_open()) {
       _socket.close();
```

Comment:
The assert here shouldn't be removed, it is not safe to close the socket concurrently, [needs to be wrapped in the strand](https://stackoverflow.com/a/35411254).